app/lib/meadow/iiif/v2/generator.ex

```diff
@@ -28,9 +28,11 @@ defmodule Meadow.IIIF.V2.Generator do
     |> Jason.encode!(pretty: true)
   end
 
+  defp to_json(manifest) when is_struct(manifest),
+    do: manifest |> Map.from_struct() |> to_json()
+
   defp to_json(manifest) do
     manifest
-    |> Map.from_struct()
     |> Enum.map(&set_property/1)
     |> Enum.filter(&filter_property/1)
     |> Enum.into(%{})
```

Comment:
After I added the metadata property map, the generator started failing because it could _only_ handle structs, not bare maps. So I moved the `Map.from_struct()` bit to a second function head. The other solution would be to create a full struct definition for properties with `label` and `value` members, but even then, `to_json/1` would still need to handle bare maps because of the language keys in both the `label` and the `value`.